encoder/CMakeLists.txt

```diff
@@ -2,8 +2,28 @@ cmake_minimum_required(VERSION 3.6)
 
 include(GNUInstallDirs)
 
-add_library(encoders encoder.cpp null_encoder.cpp h264_encoder.cpp mjpeg_encoder.cpp)
-target_link_libraries(encoders jpeg)
+set(LIBAV_PRESENT 0)
+set(SRC encoder.cpp null_encoder.cpp h264_encoder.cpp mjpeg_encoder.cpp)
+set(TARGET_LIBS images)
+
+#pkg_check_modules(LIBAV REQUIRED IMPORTED_TARGET ffspeg)
+pkg_check_modules(LIBAV QUIET IMPORTED_TARGET
+    libavcodec
+    libavdevice
+    libavformat
+)
+
+if (LIBAV_FOUND)
+    include_directories(${LIBAV_INCLUDE_DIRS})
+    set(SRC ${SRC} libav_encoder.cpp)
+    set(TARGET_LIBS ${TARGET_LIBS} ${LIBAV_LIBRARIES})
+    set(LIBAV_PRESENT 1)
+    message(STATUS "    libraries: ${LIBAV_LIBRARIES}")
+endif()
+
+add_library(encoders ${SRC})
+target_link_libraries(encoders ${TARGET_LIBS})
+target_compile_definitions(encoders PUBLIC LIBAV_PRESENT=${LIBAV_PRESENT})
```

Comment:
Our apt packages do include libav (for ffmpeg), so this will be built by default.  I wanted to make sure I did not trip over anyone else who might not have libav for whatever reason and wanted to do a custom build.